packages/headless/src/features/facets/category-facet-set/category-facet-utils.ts

```diff
@@ -33,16 +33,46 @@ export function partitionIntoParentsAndValues<
   return {parents, values};
 }
 
-export function getActiveValueFromValueTree(
-  valuesAsTrees: CategoryFacetValue[]
-): CategoryFacetValue | undefined {
+export function findActiveValueAncestry<T extends GenericCategoryFacetValue>(
+  valuesAsTree: T[]
+): T[] {
+  const ancestryMap = new Map<T, T>();
+  const activeValue = getActiveValueFromValueTree(valuesAsTree, ancestryMap);
+  return activeValue ? getActiveValueAncestry(activeValue, ancestryMap) : [];
+}
+
+function getActiveValueFromValueTree<TValue extends GenericCategoryFacetValue>(
+  valuesAsTrees: TValue[],
+  ancestryMap?: Map<TValue, TValue>
+): TValue | undefined {
   const valueToVisit = [...valuesAsTrees];
   while (valueToVisit.length > 0) {
     const currentValue = valueToVisit.shift()!;
     if (currentValue.state === 'selected') {
       return currentValue;
     }
-    valueToVisit.unshift(...currentValue.children);
+    if (ancestryMap) {
+      for (const childValue of currentValue.children) {
+        ancestryMap.set(childValue as TValue, currentValue);
```

Comment:
These cast are a bit suspicious 🤔 From what I can understand/grasp, it's because there's two types which do not perfectly overlap. These types of cast can be error prone.

I'm wondering if we could do something by changing `GenericCategoryFacetValue` to be a type where we just pick what's needed to perform the partition/identify the ancestry, instead of trying to create a type that a full union of the two.

Something like this maybe ?
```
type GenericCategoryFacetValue = Pick<
  CategoryFacetValue | CategoryFacetValueRequest,
  'state' | 'children' | 'value'
>;
```

I'm not sure that's doable, but food for thoughts.

Reply:
I lazied out a bit on this one (friday amarite), because I didn't want to refactor too much the response/request interfaces. This is now done in 827b607